pkg/api/v1/router_instance_metadata.go

```diff
@@ -116,6 +117,38 @@ func (r *Router) instanceMetadataGetInternal(c *gin.Context) {
 	}
 }
 
+// instanceMetadataExistsInternal retrieves the requested instance ID from the
+// path and looks to see if the database has metadata recorded for that ID.
+// If so, it returns a 200. If not, it returns a 404. This can be used by an
+// authenticated external system to determine which instances the metadata
+// service already knows about with minimal network overhead.
+func (r *Router) instanceMetadataExistsInternal(c *gin.Context) {
+	instanceID, err := getUUIDParam(c, "instance-id")
+
+	if err != nil {
+		invalidUUIDResponse(c, err)
+		return
+	}
+
+	metadata, err := models.FindInstanceMetadatum(c.Request.Context(), r.DB, instanceID)
+
+	if err != nil {
+		c.Status(http.StatusNotFound)
+		return
+	}
+
+	// HEAD request responses still set the Content-Length header to what it
+	// would be if we were returning the metadata
+	bytes, err := json.Marshal(metadata.Metadata)
+	if err != nil {
+		r.Logger.Sugar().Warnf("Error during json.Marshal() of metadata")
```

Comment:
Should we return a 500 error for this? Or is the default 200 response fine?

Reply:
Unrelated, but also, there's no need to call `r.Logger.Sugar()` can just call `r.Logger.Warn("...")`